inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/reduce_gpu_b_fs_yx_fsv16.cl

```diff
@@ -0,0 +1,429 @@
+                        if (fi + get_sub_group_local_id() < INPUT0_FEATURE_NUM)
+                            for (int i = 0; i < READ_OFFSET; ++i)
+                                input[i] = data[offset + get_sub_group_local_id() + i * get_max_sub_group_size()];
+                    #else
+                        input = BLOCK_READ(data, offset);
+                    #endif
+                    unroll_for (int i = 0; i < READ_OFFSET; ++i)
+                        acc = FUNC_CALL(apply_reduce)(acc, input[i]);
+                    offset += input_x_pitch * READ_OFFSET;
+                }
+                offset += input_y_pitch - (x_max_val - x_out) * READ_OFFSET * input_x_pitch;
+            }
+            offset += input_fs_pitch - ((y_max_val - y_out) * input_y_pitch);
+        }
+        offset += input_batch_pitch - ((((feature_max_val - feature_out) + FSV - 1) / FSV) * input_fs_pitch);
+    }
+
+    for (uint bi = batch_out; bi < batch_max_val; ++bi) {
+        for (uint fi = feature_out; fi < feature_max_val; fi += FSV) { 
+            for (uint yi = y_out; yi < y_max_val; ++yi) {
+                for (uint xi = x_leftover_start; xi < x_leftover_end; ++xi) {
+                    INPUT0_TYPE leftovers = INIT_VAL;
+                    #if (REDUCE_MAX_MODE || REDUCE_MIN_MODE || REDUCE_PROD_MODE || REDUCE_AND_MODE || REDUCE_LOG_SUM_EXP_MODE) && REDUCE_FEATURE && (INPUT0_FEATURE_NUM % FSV != 0)
+                        if (fi + get_sub_group_local_id() < INPUT0_FEATURE_NUM)
+                            leftovers = data[leftover_offset + get_sub_group_local_id()];
+                    #else
+                        leftovers = DT_INPUT_BLOCK_READ(data, leftover_offset);
+                    #endif
+                    acc = FUNC_CALL(apply_reduce)(acc, leftovers);
+                    leftover_offset += input_x_pitch;
+                }
+                leftover_offset += input_y_pitch - (x_leftover_end - x_leftover_start) * input_x_pitch;
+            }
+            leftover_offset += input_fs_pitch - ((y_max_val - y_out) * input_y_pitch);
+        }
+        leftover_offset += input_batch_pitch - ((((feature_max_val - feature_out) + FSV - 1) / FSV) * input_fs_pitch);
+    }
+
+    FINAL_ACCUMULATOR_TYPE final_acc;
+    acc = FUNC_CALL(sub_group_reduce)(acc);
+    final_acc = FUNC_CALL(final_reduce)(TO_FINAL_ACCUMULATOR_TYPE(acc));
+
+    OUTPUT_TYPE final_result;
+    ACTIVATION_TYPE reduce_result = TO_ACTIVATION_TYPE(final_acc);
+    #if HAS_FUSED_OPS
+        FUSED_OPS_SCALAR;
+        final_result = FUSED_OPS_RESULT_SCALAR;
+    #else
+        final_result = TO_OUTPUT_TYPE(ACTIVATION(reduce_result, ACTIVATION_PARAMS));
+    #endif
+
+    #if (REDUCE_FEATURE && REDUCE_X|| REDUCE_BATCH && REDUCE_X) && !KEEP_DIMS  
+        output[out_idx] = final_result;
+    #elif REDUCE_BX_FX_YX_X || (REDUCE_BATCH && REDUCE_Y && REDUCE_X)
```

Comment:
Using block write when there is reduction along features with keep dims (so with only one output feature) will contaminate padding. In general if there is any reduction along features sub-group has uniform values and write should happen for one work-item only.